tests/tests_app/core/test_lightning_work.py

```diff
@@ -280,3 +280,31 @@ def run(self, *args, **kwargs):
     w.run(1, [2], (3, 4), {"1": "3"})
     assert len(w._calls) == 2
     assert w._calls["0d824f7"] == {"ret": None}
+
+
+def test_work_cloud_build_config_provided():
+    class Work(LightningWork):
+        def __init__(self):
+            super().__init__()
+            self.cloud_build_config = BuildConfig(image="ghcr.io/gridai/base-images:v1.8-cpu")
+
+        def run(self, *args, **kwargs):
+            pass
+
+    w = Work()
+    assert len(w._calls) == 1
+    w.run()
+    assert len(w._calls) == 2
```

Comment:
I think these asserts do not add anything meaningful to the regression tests here, let's remove them

Reply:
Removed in commit [72be4ea](https://github.com/Lightning-AI/lightning/pull/14259/commits/72be4ea69d961eda6d16eae6e4e3f77eddb699f6)